pkg/cli/mirror/publish.go

```diff
@@ -105,60 +105,42 @@ func (o *MirrorOptions) Publish(ctx context.Context, cmd *cobra.Command, f kcmdu
 		return err
 	}
 
-	// Create backend for o.Dir
-	backend, err := storage.NewLocalBackend(o.Dir)
+	// Create a local workspace backend for incoming data
+	workspace, err := storage.NewLocalBackend(tmpdir)
 	if err != nil {
 		return fmt.Errorf("error opening local backend: %v", err)
 	}
+	// Load incoming metadta
+	if err := workspace.ReadMetadata(ctx, &incomingMeta, config.MetadataBasePath); err != nil {
+		return fmt.Errorf("error reading incoming metadata: %v", err)
+	}
 
-	// Create a local workspace backend
-	workspace, err := storage.NewLocalBackend(tmpdir)
+	// Get current metadata info
+	backendImage := fmt.Sprintf("%s/%s:latest", o.ToMirror, incomingMeta.Uid)
```

Comment:
It is more efficient to use the same repo with different tags so layers are kept within the repo. Since a uuid should be unique across the whole org at least, you can use the same repo for all `oc mirror` runs.
```suggestion
	backendImage := fmt.Sprintf("%s/oc-mirror:%s", o.ToMirror, incomingMeta.Uid)
```